### ISSUE TITLE
how to enable ipv6 only on mailtrain

### ISSUE THREAD

**Issue by ajmalkhanjamro** [Mailtrain-org/mailtrain#555]
hello i was looking for config file to modify to send through ipv6 only because i wan to experiment few things but i cant able to find any settings in zonemta or mailtrain to enable it can any buddy help

**Comment by ghost** [comment]
I'm also interested in sending from IPv6 addresses (although alongside IPv4 in my case).

Since Zone-MTA seems to support IPv6 well, I hoped that it would be a simple case of changing `ignoreIPv6` in [builtin-zone-mta.js](https://github.com/Mailtrain-org/mailtrain/blob/b4f11298ad9b0458cb822986a08961c2f8b895f5/server/lib/builtin-zone-mta.js#L133) from `true` to `false`. Although trying that just now caused errors in `zone-mta/node_modules/zone-mta/lib/sender.js`, eventually causing Mailtrain to crash.

```log
Mar 25 17:18:29 mailtrain.websmart.media node[651]: info/opt/mailtrain/zone-mta/node_modules/zone-mta/lib/sender.js:832
Mar 25 17:18:29 mailtrain.websmart.media node[651]:             delivery.connectionKey = delivery.connectionKey || [delivery.zoneAddress.address, this.zone.host || delivery.domain, delivery.mxPort].join(':');
Mar 25 17:18:29 mailtrain.websmart.media node[651]:                                                                                      ^
Mar 25 17:18:29 mailtrain.websmart.media node[651]: TypeError: Cannot read property 'address' of undefined
Mar 25 17:18:29 mailtrain.websmart.media node[651]:     at tryConnect (/opt/mailtrain/zone-mta/node_modules/zone-mta/lib/sender.js:832:86)
Mar 25 17:18:29 mailtrain.websmart.media node[651]:     at Sender.getConnection (/opt/mailtrain/zone-mta/node_modules/zone-mta/lib/sender.js:1101:9)
Mar 25 17:18:29 mailtrain.websmart.media node[651]:     at /opt/mailtrain/zone-mta/node_modules/zone-mta/lib/sender.js:740:22
Mar 25 17:18:29 mailtrain.websmart.media node[651]:     at /opt/mailtrain/zone-mta/node_modules/zone-mta/lib/sender.js:69:45
Mar 25 17:18:29 mailtrain.websmart.media node[651]:     at Immediate.<anonymous> (/opt/mailtrain/zone-mta/node_modules/zone-mta/services/sender.js:101:32)
Mar 25 17:18:29 mailtrain.websmart.media node[651]:     at processImmediate (internal/timers.js:464:21)
```

---

**Edit:** it looks like the error I was seeing was a bug in ZoneMTA that was [fixed](https://github.com/zone-eu/zone-mta/pull/277) and released in ZoneMTA v3.2.0 (2021-09-05). Mailtrain is currently locked to ^2.3.0 so is not benefiting from that fix.

**Comment by forgeOneHQ** [comment]
We are going to start with the development and testing of mailtrain v3 in the next weeks.

You are welcome to help us with the testing as soon as the first release candidate is available.
